system/lib/wasmfs/file.h

```diff
@@ -80,6 +95,9 @@ class File : public std::enable_shared_from_this<File> {
   time_t atime = 0; // Time when the content was last accessed.
 
   FileKind kind;
+
+  std::weak_ptr<File> parent; // Reference to parent of current file node that
+                              // has no ownership over the resource.
```

Comment:
The comment here just repeats what the meaning of a weak pointer is. I think it would be more useful to comment about *why* there is no ownership here.